### ISSUE TITLE
WebExtensions version?

### ISSUE THREAD

**Issue by CaptaPraelium** [benbasson/contextsearch#33]
Hi there,

Such a useful addon, it would be a shame to lose it soon. Do you have any plans for a WebExtensions version of this?

**Comment by seiferflo** [comment]
Got my vote as well ! Firefox limits to default search engine only.

**Comment by Hlsgs** [comment]
+1

**Comment by benbasson** [comment]
Sorry, it's not going to happen any time soon. Mozilla haven't implemented an API to provide access to the installed search engines, which is already 90% of the functionality of this add-on. 

https://bugzilla.mozilla.org/show_bug.cgi?id=1352598 is the blocker. According to comments, unless "someone" contributes the change, which I do not have the time or patience for, then it isn't happening any time soon.

Feel free to vote for that bug, but don't add comments unless you're willing to actually help the feature get implemented (Bugzilla isn't the place for moaning about this, but feel free to complain to Mozilla via any other means).

**Comment by benbasson** [comment]
If Mozilla manage to provide this API sometime in the next couple of months I'll revive the add-on. If it takes longer, it's possible I will lose interest (and already probably going to lose 30,000 users) and won't bother. But we'll see.

**Comment by bege10** [comment]
For those who want to stay with this extension (and the many other extensions that don't work with Firefox 57+):
Context Search 0.6.4 works with Waterfox, Context Search 0.6.3 with Palemoon.

**Comment by benbasson** [comment]
I'm sorry, I cannot agree with your recommendation.

If you're looking for an alternative browser to support this functionality, I would suggest switching to Vivaldi (which supports Chrome add-ons) or Chrome, both of which have an equivalent add-on, and have the benefit of being supported by proper companies.

Waterfox and Palemoon might be based on Mozilla source code, but I do not trust these small third-party forks enough to recommend them, e.g. in terms of security patching and reaction to zero-day vulnerabilities.

**Comment by Hlsgs** [comment]
@benbasson agreed.
I would add that, painful as it may be, the process that FF is going through is well worth the effort, if the browser is to remain relevant. And FF's userbase shouldn't abandon ship because of such minor inconveniences, but rather remain loyal, again, if the browser is to remain relevant and even gain back some of it's past importance. FF57 is already a beast performance-wise and i bet that by 59-60 most of the lost functionality will be gained back, if not all.

**Comment by benbasson** [comment]
I don't dispute that Mozilla are doing the right thing, but I do dispute the timeline. WebExtensions are not mature enough for a huge number of add-ons to be ported, but they are pushing ahead with a fixed deadline for seemingly no real reason. 

I am never likely to have one of my add-ons featured in the top 10, or maybe even the top 100, but I do know that approximately 30,000 users of this add-on are going to be waiting for an update for _months_ at best.

**Comment by Hlsgs** [comment]
@benbasson agreed on this aswell. Considering the state of the API pushing back a couple of releases would have made alot of sense. Regardless, I was just ranting about the tons of highly malicious comments i've been reading about the purported suicide of Mozilla, all laden with threats of moving to either forks of FF or Chrome :)

**Comment by benbasson** [comment]
In case I didn't make it clear enough elsewhere, I'm currently tracking (and subscribed to updates on): 

- https://bugzilla.mozilla.org/show_bug.cgi?id=1268401
- https://bugzilla.mozilla.org/show_bug.cgi?id=1352598

One or both of these needs to be resolved to my satisfaction in order to reimplement Context Search. I am hoping they will implement the suggestions in those bugs in full, but I am not holding my breath. 

Examples of things I think are (possibly) unlikely to be implemented:
- Searching with POST-based engines
- Searching in new window

**Comment by EMBBlaster** [comment]
@benbasson,
Seems that some features can be achieved. 
https://github.com/ssborbis/ContextSearch-web-ext/

**Comment by benbasson** [comment]
Hi - thanks for commenting. The author of that add-on has implemented a function to import each search engine individually, which the users have to set up above and beyond having their search engines installed in Firefox.

I specifically didn't want to do that, because the whole ethos of all my add-ons was keeping things simple. When the API is finished, I will do what I can and update this ticket. 

To be frank, the only reason I'm still here offering to do this is because of the 14 years of loyal usage and friendly feedback I've had from a few tens of thousands of people. This doesn't make me any money, and I no longer use my own add-ons, or Firefox at all. In other words, while the add-on you point to looks excellent, that's about 99% more work than I am ever going to put in.

**Comment by EMBBlaster** [comment]
> Hi - thanks for commenting.

No, I'm thanking you for your work and time devoted to this. Being working with open source code myself, I really appreciate what you did.

> The author of that add-on has implemented a function to import each search engine individually, which the users have to set up above and beyond having their search engines installed in Firefox.
I specifically didn't want to do that, because the whole ethos of all my add-ons was keeping things simple - in terms of user experience and also for me to maintain. When the API is finished, I will do what I can and update this ticket.

Yes. I agree. Too much work and not really simple. 
As I didn't know if you was aware of that extension, I leaved that comment in hope it could be of any help for you.

**Comment by evilpie** [comment]
Hey! I just wanted to let you know that [bug 1352598](https://bugzilla.mozilla.org/show_bug.cgi?id=1352598) just landed in Nightly. @smile4ever even already started documenting this on [MDN](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/

**Comment by Smile4ever** [comment]
@evilpie It didn't land yet in the Nightly builds, I guess we need to have a bit more patience 😀

**Comment by jtojnar** [comment]
https://github.com/tiansh/simple-context-search looks like a good alternative.

**Comment by Millichrome** [comment]
@jtojnar 
Thanks for the link. It's almost a perfect alternative to Basson's context search. 
One thing I don't like is that it creates a new context menu (although it does support existing search providers out of the box); you still get the "Search [XXX] in Google" context menu. It also doesn't quote the actual text selection. The latter issue seems to be limitation with FF's WebExtension engine.

**Comment by Chealer** [comment]
This looks like a duplicate of ticket #27.

**Comment by psxlover** [comment]
@Chealer why would you comment on a ticket this old on a repo that has not seen any changes for 9 years?

If you are looking for an alternative, don't know what else is there but I'm currently using https://github.com/tiansh/simple-context-search
